CDP4Composition/Composition/CDP4Bootstrapper.cs

```diff
@@ -0,0 +1,170 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="CDP4Bootstrapper.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2021 RHEA System S.A.
+//
+//    Author: Sam Gerené, Alex Vorobiev, Alexander van Delft, Nathanael Smiechowski, Simon Wood
+//
+//    This file is part of CDP4-IME Community Edition.
+//    The CDP4-IME Community Edition is the RHEA Concurrent Design Desktop Application and Excel Integration
+//    compliant with ECSS-E-TM-10-25 Annex A and Annex C.
+//
+//    The CDP4-IME Community Edition is free software; you can redistribute it and/or
+//    modify it under the terms of the GNU Affero General Public
+//    License as published by the Free Software Foundation; either
+//    version 3 of the License, or any later version.
+//
+//    The CDP4-IME Community Edition is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//    GNU Affero General Public License for more details.
+//
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program. If not, see <http://www.gnu.org/licenses/>.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace CDP4Composition.Composition
+{
+    using System;
+    using System.ComponentModel.Composition;
+    using System.ComponentModel.Composition.Hosting;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Reflection;
+
+    using CDP4Composition.Modularity;
+    using CDP4Composition.Services.AppSettingService;
+
+    using Microsoft.Practices.ServiceLocation;
+
+    using NLog;
+
+    /// <summary>
+    /// Base class that provides basic bootstrapping using MEF
+    /// </summary>
+    /// <typeparam name="T">The <see cref="AppSettings"/></typeparam>
+    public abstract class CDP4Bootstrapper<T> where T : AppSettings, new()
+    {
+        /// <summary>
+        /// A <see cref="Logger"/> instance
+        /// </summary>
+        protected static Logger logger = LogManager.GetCurrentClassLogger();
+
+        /// <summary>
+        /// Represents the state of the <see cref="CDP4Bootstrapper"/>
+        /// </summary>
+        protected string status;
+
+        /// <summary>
+        /// Runs the bootstrapper
+        /// </summary>
+        public void Run()
+        {
+            UpdateBootstrapperStatus("Configuring catalogs");
```

Comment:
please use explicit `this.` qualifier here and below and in the other methods of this file.